src/chart/components/volumes/volumes.component.ts

```diff
@@ -136,18 +151,21 @@ export class VolumesComponent extends ChartBaseElement {
 		this.canvasModel.fireDraw();
 	}
 
-	private addVolumesToDynamicObjects() {
+	private changeVolumesDynamicObject() {
+		const visible = this.config.components.volumes.visible;
+
+		if (!visible) {
+			this.dynamicObjectsComponent.model.removeObject(this.volumesModel.id);
+			return;
+		}
+
 		// check if the volumes dynamic object is already added
 		const position = this.dynamicObjectsComponent.model.getObjectPosition(this.volumesModel.id);
-		if (position !== -1) {
+		if (position === -1) {
+			this.dynamicObjectsComponent.model.addObject(this.volumesDynamicObject);
```

Comment:
Maybe it is better for code reading to restore an object destructure? Because it is difficult to track where paneId is changed.

Or we can make a setter before adding object to ` this.dynamicObjectsComponent.model` and after that use `this.volumesDynamicObject`

But, if `this.config.components.volumes.visible === false`,  we delete it from `this.dynamicObjectsComponent.model`, but we still have `this.volumesDynamicObject` in memory.
In general, looks like it is unnecessarily to keep it in this class, and use getter/setter for it